src/Views/OperatingSystemView.vala

```diff
@@ -95,40 +103,116 @@ public class About.OperatingSystemView : Gtk.Box {
 
-        Gtk.Button? update_button = null;
-        var appcenter_info = new GLib.DesktopAppInfo ("io.elementary.appcenter.desktop");
-        if (appcenter_info != null) {
-            update_button = new Gtk.Button.with_label (_("Check for Updates"));
-            update_button.clicked.connect (() => {
-                appcenter_info.launch_action ("ShowUpdates", new GLib.AppLaunchContext ());
-            });
-        }
+        updates = new Gtk.StringList (null);
+
+        var update_list = new Gtk.ListBox () {
+            vexpand = true,
+            selection_mode = Gtk.SelectionMode.SINGLE
+        };
+        update_list.bind_model (updates, (obj) => {
+            var str = ((Gtk.StringObject) obj).string;
+            return new Gtk.Label (str);
+        });
+
+        var update_scrolled = new Gtk.ScrolledWindow () {
+            child = update_list
+        };
+
+        updates_image = new Gtk.Image () {
+            icon_size = LARGE,
+            margin_end = 6
+        };
+
+        updates_title = new Gtk.Label (null) {
+            hexpand = true,
+            margin_end = 6,
+            xalign = 0
+        };
+
+        updates_description = new Gtk.Label (null) {
+            xalign = 0
+        };
+        updates_description.add_css_class (Granite.STYLE_CLASS_SMALL_LABEL);
+        updates_description.add_css_class (Granite.STYLE_CLASS_DIM_LABEL);
+
+        var update_button = new Gtk.Button.with_label (_("Download")) {
+            margin_end = 6,
+            valign = CENTER
+        };
+        update_button.add_css_class (Granite.STYLE_CLASS_SUGGESTED_ACTION);
+
+        update_button_revealer = new Gtk.Revealer () {
+            child = update_button,
+            overflow = VISIBLE,
+            transition_type = SLIDE_LEFT
+        };
+
+        var cancel_button = new Gtk.Button.with_label (_("Cancel")) {
+            margin_end = 6,
+            valign = CENTER
+        };
+
+        cancel_button_revealer = new Gtk.Revealer () {
+            child = cancel_button,
+            overflow = VISIBLE,
+            transition_type = SLIDE_LEFT
+        };
+
+        var error_button = new Gtk.Button.with_label (_("Refresh")) {
+            margin_end = 6,
+            valign = CENTER
+        };
+
+        error_button_revealer = new Gtk.Revealer () {
+            child = error_button,
+            overflow = VISIBLE,
+            transition_type = SLIDE_LEFT
+        };
+
+        var updates_grid = new Gtk.Grid () {
+            margin_top = 6,
+            margin_bottom = 6,
+            margin_start = 6
+        };
+        updates_grid.attach (updates_image, 0, 0, 1, 2);
+        updates_grid.attach (updates_title, 1, 0);
+        updates_grid.attach (updates_description, 1, 1);
+        updates_grid.attach (update_button_revealer, 2, 0, 1, 2);
+        updates_grid.attach (cancel_button_revealer, 3, 0, 1, 2);
+        updates_grid.attach (error_button_revealer, 4, 0, 1, 2);
```

Comment:
Maybe if we're gonna have a button for every state we should put them in a stack :joy:

Reply:
Would probably make sense :)